core/src/main/java/org/elasticsearch/action/support/broadcast/BroadcastResponse.java

```diff
@@ -71,6 +74,17 @@ public int getFailedShards() {
     }
 
     /**
+     * The REST status that should be used for the response
+     */
+    public RestStatus getStatus() {
+        if (failedShards > 0) {
+            return RestStatus.INTERNAL_SERVER_ERROR;
```

Comment:
shall we make an heroic effort to take the worst of all underlying rest status? maybe just the first ?